### ISSUE TITLE
Merge ofxiOS and ofxAndroid where possible

### ISSUE THREAD

**Issue by HalfdanJ** [openframeworks/openFrameworks#4612]
Some thoughts, i'm thinking it would be great to merge parts of ofxiOS and ofxAndroid together. They have a lot in common, and it would be great if creating an app for iOS would be rather easy to port to android, and vice versa. 
A goal would be that most of the examples in iOS examples folder should also be able to run on Android.

Currently they both have each their own addon-ish baseapp sub-class, could we merge their commons into a ofBaseMobileApp class in the core instead? The naming in the two classes are not the same, so it requires some renaming, but i think for the better. 

By doing this it might also help with better differentiation between core code for  non mobile and mobile devices, using the namespace ofBaseMobil. 

In my opinion now would be a good time to do it. ofxAndroid is going through a larger refactor with #4260, and it would be great with more examples for it.

**Comment by julapy** [comment]
im kind of divided on this suggestion.
a part of me always wants to standardise the API, but at the same time i feel theres still loads of changes happening in Android and iOS, and trying to create a common API for both at this stage, might stager development.

i have seen little things that should be standardised, like for example the double tap event is fired at different stages on iOS and Android. i can't remember which way around it is now, but i think on iOS when you double tap, you receive a touchDownEvent, a touchDoubleTapEvent and then a touchDownEvent. its different on Android... and not sure which one is correct, but something like this should be standardised.

maybe this issue can become a thread for discussion which things need to be standardised across Android and iOS.

**Comment by danthemellowman** [comment]
I don't like this. I want to move to a completely native approach on
android similar to what cinder just released.

On Sunday, December 6, 2015, lukasz karluk <notifications@github.com> wrote:

> im kind of divided on this suggestion.
> a part of me always wants to standardise the API, but at the same time i
> feel theres still loads of changes happening in Android and iOS, and trying
> to create a common API for both at this stage, might stager development.
>
> i have seen little things that should be standardised, like for example
> the double tap event is fired at different stages on iOS and Android. i
> can't remember which way around it is now, but i think on iOS when you
> double tap, you receive a touchDownEvent, a touchDoubleTapEvent and then a
> touchDownEvent. its different on Android... and not sure which one is
> correct, but something like this should be standardised.
>
> maybe this issue can become a thread for discussion which things need to
> be standardised across Android and iOS.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openframeworks/openFrameworks/issues/4612#issuecomment-162357655>
> .
>


-- 
Sent from Gmail Mobile #autocorrect

**Comment by HalfdanJ** [comment]
Hey @danthemellowman, can you explain a bit more? i don't see why a unified baseapp for iOS and android is standing in the way for the android core backend. I see it as a way to make sure the naming is the same, when you write an ofApp.

**Comment by arturoc** [comment]
Yes i don't think those have anything to do, we could have a native activity and still have some common bases between ios and android

**Comment by danthemellowman** [comment]
I must have misread what you guys are talking about.


On Monday, December 7, 2015, arturo <notifications@github.com> wrote:

> Yes i don't think those have anything to do, we could have a native
> activity and still have some common bases between ios and android
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openframeworks/openFrameworks/issues/4612#issuecomment-162488728>
> .
>


-- 
Sent from Gmail Mobile #autocorrect
